### DISCUSSION TITLE
For those who have access to flags, incognito

### DISCUSSION THREAD

**Discussion by s0urce-c0de** [3kh0/ext-remover#1100] [Fixes JOIN TITANIUM NETWORK TO SHARE PLEASE]
<details>
<summary>https://crbug.com/341245382 (not by me, i did not discover this)</summary>
DESCRIPTION jo...@gmail.com created issue <a href="https://issues.chromium.org/issues/341245382#comment1">#1</a><div style="text-align:right;">May 17, 2024 06:46AM</div>

Security Bug

Important: Please do not change the component of this bug manually.

Please READ THIS FAQ before filing a bug: https://chromium.googlesource.com/chromium/src/+/HEAD/docs/security/faq.md

Please see the following link for instructions on filing security bugs: https://www.chromium.org/Home/chromium-security/reporting-security-bugs

Reports may be eligible for reward payments under the Chrome VRP: https://g.co/chrome/vrp

NOTE: Security bugs are normally made public once a fix has been widely deployed.

-------------------------

VULNERABILITY DETAILS
On Chrome OS, you can open an Incognito browsing window through the flag #captive-portal-popup-window even when the policy IncognitoModeAvailability equals 1 which should disable Incognito mode.

VERSION
Chrome Version: Tested on 124.0.6367.154 (Official Build) (64-bit), but the code path is still present in the HEAD on Chromium Code Search.
Operating System: Chrome OS 124.0.6367.154 (Official Build) (64-bit)

REPRODUCTION CASE
1. Ensure Incognito is disabled by IncognitoModeAvailability = 1.
2. Enable #captive-portal-popup-window in flags.
3. Trigger a captive portal notification by connecting to a network with a captive portal or by setting the DNS to one which triggers a captive portal.
4. Click 'Sign in' to open the captive portal popup.
5. With the captive portal popup focused, press Control-T. This will open a new incognito window, bypassing policy. You can now close the captive portal popup and switch to a working DNS or network.
NOTE: You can also right-click -> View Source if DeveloperToolsAvailability allows such.

CODE PATH
The main problem is [here](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/ash/network/network_portal_signin_controller.cc;drc=2896d2a0a69319ecbb68c12ecd25638640b0d923;l=176). It explicitly checks if Incognito is disabled by policy, but if it is, it runs the same code as if not. The comment says:
Since the signin window enables extensions and disables navigation, no special handling is required when Incognito browsing is disabled by policy.
But the function it calls (definition [here](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/chromeos/network/network_portal_signin_window.cc;drc=c0265133106c7647e90f9aaa4377d28190b1a6a9;l=70)) uses an OTR profile regardless of IncognitoModeAvailability, and does not enable extensions as the comment asserts. To fix the problem, I would change the first file (chrome/browser/ui/ash/network/network_portal_signin_controller.cc) to not open the popup window if Incognito is disabled, as even if only the Control-T is disabled as a "fix", one can still run their own DNS server which will display something like Google in the captive portal, and bypass filtering.

FOR CRASHES, PLEASE INCLUDE THE FOLLOWING ADDITIONAL INFORMATION
N/A

CREDIT INFORMATION
Externally reported security bugs may appear in Chrome release notes. If this bug is included, how would you like to be credited?
Reporter credit: @️&ZeroWidthSpace;joshuajohncohen (GitHub)
![example](https://issues.chromium.org/action/issues/341245382/attachments/56445105?download=false)
</details>

Instructions for skids:
* go to chrome://flags/#captive-portal-popup-window
* if it didn't work, UPGRADE CHROME
* enable it
* restart
* hope your flags are still there. if they are continue, else you can't set flags
* use a dns which gives you a captive portal
* click CTRL+T after clicking sign in
* INCOGNITO MODE
* if it didn't work, might get patched.

**EDIT:**
THIS IS NOT MY SCHOOL, THIS IS THE IMAGE FROM THE [BUG REPORT](https://crbug.com/341245382) I ATTACHED.

If this gets patched, it is not my fault. This is listed as a Severity 4 Priority 4 Security Bug on the bug reporter.
I don't know if this is the correct dns, but lots of people say you can use `detectportal.firefox.com` as the DNS

**Comment by S-PScripts** [comment]
looks like the incognito exploit is back but easier?!

**Reply by S-PScripts** [reply]
what dns would open a captive hmm

**Comment by S-PScripts** [comment]
i think you doxxed your school i think you should edit and delete the first revision

**Comment by S-PScripts** [comment]
IT WORKS

**Reply by S-PScripts** [reply]
THE INCOGNITO EXPLOIT IS BACK

DNS TO USE: detectportal.firefox.com (from https://github.com/jee1mr/captive-portal)

**Reply by S-PScripts** [reply]
Also unblocks extensions!!!!!!!!!!!!!!!!!!! just like the original one

also make sure to change name server back to automatic

**Reply by akabutnicer** [reply]
since when could it ever do this?

**Comment by S-PScripts** [comment]
uh it opened the old incognito exploit captive tab too (separately) lol

i tried using the old incognito exploit DNS (52.207.185.90) but it didn't work

i guess it did but i didn't get the sign in popup??

**Comment by S-PScripts** [comment]
![image](https://github.com/3kh0/ext-remover/assets/136199723/81e4b17b-a047-4d75-aad7-534aeb15d0cb)

**Reply by S-PScripts** [reply]
the right was unintentional since i didnt notice , the left was what i did

**Reply by S-PScripts** [reply]
@AshtonDavies

**Reply by S-PScripts** [reply]
@Brandon421-ops

**Comment by S-PScripts** [comment]
**Here's a version I wrote on this** 
The incognito exploit is BACK! (v124)

Instructions:
1. Go to chrome://flags/#captive-portal-popup-window.
2. If it doesn't exist, make sure to update chromeOS to the latest version.
3. Enable it.
4. Restart.
5. If the flag didn't reset, you can continue. Else you cannot.
6. Go to Settings.
7. Click Wifi in the Network section.
8. Click your wifi.
9. Click Network.
10. Set Name servers to Custom name servers.
11. Set the first box to detectportal.firefox.com.
12. A sign in pop up should appear from your wifi. Click Sign in.
13. Do CTRL+T - YOU'RE NOW IN INCOGNITO MODE!
14. Set Name servers back to what it was before.

Notes:
-> Extensions can't block you!
-> No search history!

Credits:
https://crbug.com/341245382
https://github.com/s0urce-c0de for finding the post

**Reply by S-PScripts** [reply]
https://github.com/S-PScripts/chromebook-utilities/blob/main/Exploits/Incognito%20(v124)

**Reply by S-PScripts** [reply]
https://s-pscripts.github.io/incognito-v124/

**Reply by op-max** [reply]
I found a similar way to access incognito via WiFi. But I'm still confused on some parts of it.

**Reply by S-PScripts** [reply]
ok

**Reply by S-PScripts** [reply]
> I found a similar way to access incognito via WiFi. But I'm still confused on some parts of it.

I've never got the thing you said happen over in your issue

https://s-pscripts.github.io/incognito-v123/ is the new link for my incognito btw since the one on may 31 was changed

**Reply by s0urce-c0de** [reply]
oh ok that is why the old link didn't work

**Reply by op-max** [reply]
Perhaps the reason it does not work for most people is the network type. Different companies like Verizon manage DNS servers differently.

**Reply by S-PScripts** [reply]
> Perhaps the reason it does not work for most people is the network type. Different companies like Verizon manage DNS servers differently.

note: i did find the incognito via wifi actually, its in the link (method 2) https://s-pscripts.github.io/incognito-v123/ . However, it's not exactly the same as yours.

i used ee wifi

**Comment by S-PScripts** [comment]
lots of people xd its just me yapping

ngl why are more people not actually talking about this- it's actually good

**Reply by s0urce-c0de** [reply]
what is the name/DNS server

**Reply by S-PScripts** [reply]
If you meant as to what kind, that would be Custom name servers

If you were talking about the name/DNS though, I told you that, you even added it to your post! It's detectportal.firefox.com (you don't need an IP address for custom name servers)

**Reply by s0urce-c0de** [reply]
On my chromebook it says no internet

**Reply by s0urce-c0de** [reply]
i just added it since everybody said it was the DNS server

**Reply by S-PScripts** [reply]
> On my chromebook it says no internet

you need to do this:
-> in settings, go to the network section
-> click your wifi
-> click your wifi again under the known networks section
-> go to the network section on your wifi
-> click it
-> set name servers to custom
-> set all the boxes to detectportal.firefox.com. if no sign in popup shows up, try disconnecting and reconnecting  and trying again.
-> a signin popup shows up, click sign in on that popup
-> do ctrl + t to open a new tab
-> set your name servers back to automatic

**Reply by S-PScripts** [reply]
_you have to change your name servers back after doing ctrl + t, otherwise your wifi won't work._

**Reply by s0urce-c0de** [reply]
No, it doesn't give me a portal, only a No Internet. Also, this exploit doesn't work on my chromebook because I don't have flags. I was just trying to see if it works.

**Reply by S-PScripts** [reply]
> No, it doesn't give me a portal, only a No Internet. Also, this exploit doesn't work on my chromebook because I don't have flags. I was just trying to see if it works.

damn no flags? also you probably don't have a portal thing since the flag is probably required in some way (idk what)

**Reply by S-PScripts** [reply]
Captive portal signin uses popup window.
Use a popup window with proxies disabled for captive portal signin. – ChromeOS

hmm

**Reply by s0urce-c0de** [reply]
@S-PScripts My chromebook goes to a sign in screen where you have to enter your email each time i restart. I could give you the values of some policies if you need it.
My mandatory certificates is:
```
{\n   \"Certificates\": [ {\n      \"GUID\": \"{4a4d31ca-c2a1-4f40-b16c-329dd0f0c1c8}\",\n      \"TrustBits\": [ \"Web\" ],\n      \"Type\": \"Authority\",\n      \"X509\": \"-----BEGIN CERTIFICATE-----\\nMIIDdzCCAl+gAwIBAgIJAOu3HNNoWOxSMA0GCSqGSIb3DQEBCwUAMFIxCzAJBgNV\\nBAYTAlVTMRMwEQYDVQQIDApDYWxpZm9ybmlhMRYwFAYDVQQKDA1TZWN1cmx5LCBJ\\nbmMuMRYwFAYDVQQDDA0qLnNlY3VybHkuY29tMB4XDTE1MDEyNjIzMjkzOVoXDTIw\\nMDEzMDIzMjkzOVowUjELMAkGA1UEBhMCVVMxEzARBgNVBAgMCkNhbGlmb3JuaWEx\\nFjAUBgNVBAoMDVNlY3VybHksIEluYy4xFjAUBgNVBAMMDSouc2VjdXJseS5jb20w\\nggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQC1NJCxT8I0Muf3bDGeBXo2\\nX0t7z4ZziKJmJUpOaeR48b+343bKvlg20o097GJopSnvi9rZZopzWVo3oVIkgs96\\nm4DpQE+ShyAXNbnjQuuYb4vjGFJ130pZn9O3wlT2WLJhqn3ojYv49eKzXG3IuqYy\\nAxyHgURoGB8hEjgxWkGA8k5/BpxmE1RTXQCl5X+hBIB/kQNzMzoKM2jI5nwhU8kj\\n5o/D4X1a1RmfNl08u0JsMw+dFG9BzletiXdk8QaNnRg4KoSDb1Pl2RRxEAxzm/pY\\n72rzRWh/AVjd5qbP3Rs7VDX1MQYVGol5ThEVH6cQH+xwJyZar3c4SqnCxSV/oAuv\\nAgMBAAGjUDBOMB0GA1UdDgQWBBQW39BDC8Drcbo7+nmhvKMr8uPUdTAfBgNVHSME\\nGDAWgBQW39BDC8Drcbo7+nmhvKMr8uPUdTAMBgNVHRMEBTADAQH/MA0GCSqGSIb3\\nDQEBCwUAA4IBAQCZ1xAuCLrObR08XLr0fXbvazp+IZidemjb82z/eWeqdxmN5hCU\\nAScMhszPlc8ySB66A5Uf8EoWCbuebmpf9LX5b4d88Ka6Xs2Ki6LO6SWkt6nNVwRJ\\n7oFHJRKvLw+GDbR37YVxRkvKPZxeklc0X6dVmgUSGXrIVy2WSd5oIySKygH6K/Th\\n8RvMWitOgXrIc5Lw8yYHgQvx8V2kU0axU/AjY28kRpi5iKfF251HpJuSrTD431GR\\nQc1VnYid++ge80JJX+fQuw+DUi9s4DIbNiPczP6leh+DYpc0gcoHmaO8SXHU/jFG\\nOonUU7nUehYOAILUkHN87zHrZ5N99QWi3sRi\\n-----END CERTIFICATE-----\"\n   } ]\n}\n
```
URLBlocklist:
```
[ "chrome://history-frame", "chrome://chrome/history-frame", "javascript://*" ]
```
Allowed Extensions:
```
[ "haldlgldplgnggkjaafhelgiaglafanh", "jcocgejjjlnfddlhpbecfapicaajdibb", "knhlokifbjpnnihogpbdmblmkadbfjfm", "aapbdbdomjkkjkaonfhkkikfgjllcleb", "aedgoefpkibhaiigkbjjgccgncldoice", "ahmapmilbkfamljbpgphfndeemhnajme", "ajiejmhbejpdgkkigpddefnjmgcbkenk", "aohghmighlieiainnegkcijnfilokake", "bfpeiapdhnegnfcfkdfihabadngjagfj", "bijihlabcfdnabacffofojgmehjdielb", "bjkfigedpppnggdamgeclieeekhbgjpc", "cahedbegdkagmcjfolhdlechbkeaieki", "ccpkbmhndgcecbbmimeomchakbadhgbp", "cdnapgfjopgaggbmfgbiinmmbdcglnam", "cgbbbjmgdpnifijconhamggjehlamcif", "chphlpgkkbolifaimnlloiipkdnihall", "cjpalhdlnbpafiamejdnhcphjbkeiagm", "dhgmfcabdnkbdhelnooodefedbilcpho", "dikiaagfielfbnbbopidjjagldjopbpa", "djnhkfljnimcpelfndpcjcgngmefaobl", "dncgedbnidfkppmdgfgidcepclnokpkb", "eekbbmglbfldjpgbmajenafphnfjonnc", "eepfnpdggadinkhdkeeclnodfgnppheh", "fbcgkphadgmbalmlklhbdagcicajenei", "gbkeegbaiigmenfmjfclcdgdpimamgkj", "ghbmnnjooekpmoecnnnilnnbdlolhkhi", "hdcpibgmmcnpjmmenengjgkkfohahegk", "hdhinadidafjejdhmfkjgnolgimiaplp", "hdjklakffhgbjmenlcjcgagjlkbpmjcf", "hfhhnacclhffhdffklopdkcgdhifgngh", "hmbjbjdpkobdjplfobhljndfdfdipjhg", "hodndjmaknljbbjpmmpffpidoahkgibh", "ibpndfoeinanpbdncikcodamekdjpkea", "icmaknaampgiegkcjlimdiidlhopknpk", "idmlfdpopnofadegehbbknejjnejeinm", "ifkgpacemihiplnocjocpgmoiefcojik", "iipifbplbkfglpemmbhmaijjlknkaiih", "ijnjlojbdhgpamjiflocklhfeciokfdl", "inoeonmfapjbbkmdafoankkfajkcphgd", "iomokcfebnfiflpgcpcijfkfmafgkjgh", "ipldiafffkbheamaoepjlncolocdlojd", "jmmkbcfakiendpnceenfanaebjenjene", "kbbakflmccgdckcccapdedjpcemkpoeo", "kbfnbcaeplbcioakkpcpgfkobkghlhen", "kfokdmfpdnokpmpbjhjbcabgligoelgp", "kggefgejofekboiejlbaaodipdplabpc", "kneikgicddfadmnpnjhbfiaoomchignh", "knibbpmgfkaginjkddpmmpcnknegecok", "laookkfknpbbblfpciffpaejjkokdgca", "lkllajgbhondgjjnhmmgbjndmogapinp", "lpakoahdokkkonadfppfgmednkknpgbm", "lpcaedmchfhocbbapmcbpinfpgnhiddi", "mamepagkigmnpoalafajabnljlkkocbk", "mapfkcmnklanficcnnjkgeneakedmjkp", "mgneogobocdccfjbdldekbjfdadhncol", "mmeijimgabbpbgpdklnllpncmdofkcpn", "naajkjmkhbpchkbhbcecjojhalhdddjg", "nelnpbigggagbdcfehjcnfbgaleelhop", "nfiekeidmabgihoefbofbeepmcdglhji", "nhkoenoennbjnibepkjdheodiaojdgpk", "nopfnnpnopgmcnkjchnlpomggcdjfepo", "oglcbfocdkemachifmnimbblndgmlhbi", "ohcpnigalekghcmgcdcenkpelffpdolg", "oipimoeophamdcmjcfameoojlbhbgjda", "okecofhlheppfglpgakbmhhohicoikcm", "okgjbfikepgflmlelgfgecmgjnmnmnnb", "okiblndpcefmebnkjnjfplijnelbcjmm", "oohcehjlgohbhbljoepeoedbgcpcefod", "pejkokffkapolfffcgbmdmhdelanoaih", "peocoajdkpphnafgbgcieeclnlahlcae", "pkehgijcmpdhfbdbbnkijodmdjhbjlgp", "plcopbelnjbdinpapnoibincnedalekm", "ajphlblkfpppdpkgokiejbjfohfohhmk", "bdakmnplckeopfghnlpocafcepegjeap", "bfgdeiadkckfbkeigkoncpdieiiefpig", "bgdpkilkheacbboffppjgceiplijhfpd", "bgofflgeghkhocbociocnckocbjmomjh", "ccachieepekabdbcpfegamckdidpepga", "cdochbecpfdpjobpgnacnbepkgcfhoek", "cfhdojbkjhnklbpkdaibdccddilifddb", "cjanmonomjogheabiocdamfpknlpdehm", "cmedhionkhpnakcndndgjdbohmhepckk", "dakebdbeofhmlnmjlmhjdmmjmfohiicn", "djonnbgfieijldcieafgjcnhmpcfpmgg", "dnloadmamaeibnaadmfdfelflmmnbajd", "dpgjihldbpodlmnjolekemlfbcajnmod", "ecldhagehndokdmaiaigoaecbmbnmfkc", "ecnphlgnajanjnkcmbpancdjoidceilk", "eiimnmioipafcokbfikbljfdeojpcgbh", "enebomhlllfaccbelnjhfgblnalofhch", "feepmdlmhplaojabeoecaobfmibooaid", "fhggacdeldojnpbgknpipalghlkbcimk", "fmidkjgknpkbmninbmklhcgaalfalbdh", "gcoekeoenehjmndhkdnoomdjeaclkhbe", "gdicnpbaekbefjanokchpfhnaphfnphl", "hjngolefdpdnooamgdldlkjgmdcmcjnc", "ifajfiofeifbbhbionejdliodenmecna", "iljojpiodmlhoehoecppliohmplbgeij", "ipkjmjaledkapilfdigkgfmpekpfnkih", "kcnhkahnjcbndmmehfkdnkjomaanaooo", "kgjfgplpablkjnlkjmjdecgdpfankdle", "kkkbiiikppgjdiebcabomlbidfodipjg", "kohfgcgbkjodfcfkcackpagifgbcmimk", "laankejkbhbdhmipfmgcngdelahlfoji", "lhiccpgcnopcjjdobhoddnplkebplfaj", "lmepjnndgdhcgphilomlfekmgnnmngbi", "mfedioikeigljhjfpghdejnogniddhna", "mhccomhlpeimdkagphppaoefhkeopnho", "mhpdeoicbejhlhbljpaadliigmifdajh", "mjgcgnfikekladnkhnimljcalfibijha", "mloajfnmjckfjbeeofcdaecbelnblden", "nijejdnikeoaldbcboagjlibadkabiae", "nlipoenfbbikpbjkfpfillcgkoblgpmj", "ohfgljdgelakfkefopgklcohadegdpjf", "ohfjebjepnlldifcbcfmopifaebcjehc", "oikgofeboedgfkaacpfepbfmgdalabej", "omebobahbkampglebglkoagddjnjbhle", "ppckapbnfhikdajgehibjapcohbaomhd" ]
```
I have goguardian.

**Reply by S-PScripts** [reply]
is IncognitoModeAvailability set to 1

**Reply by S-PScripts** [reply]
i have pretty much the same blocklist and i also get a log in screen when restarting (or locking)

**Reply by S-PScripts** [reply]
you probably have a policy that blocks chromeflags. chromeflags has that flag which you need for this exploit

you probably also need to be able to have incognito mode availability set so its actually possible to use in some way apart from chromeflags

**Reply by s0urce-c0de** [reply]
where is that policy?

**Reply by s0urce-c0de** [reply]
my login screen i need internet+email+password

**Reply by S-PScripts** [reply]
my login screen has the wifi already on and email set

i just put the password

not sure what policy flags is

**Reply by s0urce-c0de** [reply]
probably a device flag

**Reply by S-PScripts** [reply]
yes

**Comment by brobcoolguy** [comment]
does this still work? im on v126 and after activating the flag the signin popup doesnt open in a seperate portal, just in a chrome tab

**Reply by S-PScripts** [reply]
im on v127 and it somehow works still

**Reply by wvminecraftkid** [reply]
me too it might be because the school uses a proxy

**Comment by ghost** [comment]
no longer works on 128 :(

**Reply by S-PScripts** [reply]
indeed

**Reply by akabutnicer** [reply]
indeed

**Reply by AshtonDavies** [reply]
indeed
